### PR TITLE
Implement images-sync

### PR DESCRIPTION
**Pull request by gboutry**
Add the `images-sync` plugin, enabling sync images from a SimpleStreams source.

Requires:
- [x] https://github.com/canonical/sunbeam-terraform/pull/64

### CODE REVIEW

**Comment by gboutry** [comment]
Rebase on main + add manifest entries to default manifests

File: sunbeam-python/sunbeam/plugins/images_sync/plugin.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        """Enable images-sync service."""
+        super().enable_plugin()
+
+    @click.command()
+    def disable_plugin(self) -> None:
+        """Disable images-sync service."""
+        super().disable_plugin()
+
+    @property
+    def k8s_application_data(self):
```
Comment (hemanthnakkina) [inline sunbeam-python/sunbeam/plugins/images_sync/plugin.py:95]: This function is not updated in all the plugins. We need to revisit if this is required or not.

**Review by hemanthnakkina — APPROVED** [review]

File: sunbeam-python/sunbeam/plugins/images_sync/plugin.py (hunk 1 above)
Comment (gboutry) [reply sunbeam-python/sunbeam/plugins/images_sync/plugin.py:95]: Is this still needed since we have manifest ?

Comment (hemanthnakkina) [reply sunbeam-python/sunbeam/plugins/images_sync/plugin.py:95]: I dont think this is required. We can remove it as separate PR from this plugin and dns plugin and plugin base.

**Pull request merged**